x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_columns/use_columns.ts

```diff
@@ -169,17 +169,6 @@ export const useColumns = ({
     [defaultColumns]
   );
 
-  useEffect(() => {
-    // if defaultColumns have changed,
-    // get the latest columns provided by client and
-    if (didDefaultColumnChange) {
-      defaultColumnsRef.current = defaultColumns;
-      setColumnsPopulated(false);
```

Comment:
I think I understand the problem and it seems to be 2 pronged.

1. Trigger Actions alert table does natively supported changing columns from the client side. So I added a work around, that we will provide a unique set of columns `per ID` and that is how we achieve columns difference between Event Rendered View v/s Grid View.
   - Now here, for a single ID ( i.e. gridView ) we are changing the columns for grid View from the client side (i.e. security solution ) which is causing weird issues even if I add above line : `setColumnsPopulated(false)`
 
2. Secondly, single source of truth on the storage is maintained by `Triggers action alert table`. Changing security solution columns state does will not have impact because if we add a field from field browser, it will only change `trigger-actions` state of columns not security solution `columns` state. Currently, there is no sync.
 

Currently, only safe way seems to me is to get handle `onToggleColumn` function, create a context in `Security Solution` and then use that toggle function anywhere to add columns to the table.  This is because trigger actions table wants to handle how the columns are changed in the table. 

Please let me know if something does not make sense.. we can have a meeting.

Reply:
Thanks for the write up, it does make sense. Took a second pass at it if you have some time!